proxy/ratelimit_test.go

```diff
@@ -232,74 +92,15 @@ func TestCheckLocalRateLimit(t *testing.T) {
 	}, r...)
 	defer p.Close()
 
-	doc := func(l int) []byte {
-		b := make([]byte, l)
-		n, err := rand.Read(b)
-		if err != nil || n != l {
-			t.Fatal("failed to generate doc", err, n, l)
-		}
-
-		return b
-	}
-
-	request := func(doc []byte) (int, http.Header) {
-		req, err := http.NewRequest("GET", p.URL+"/", nil)
-		if err != nil {
-			t.Fatal("foo", "failed to create request", err)
-			return -1, nil
-		}
-
-		req.Close = true
-
-		rsp, err := (&http.Client{}).Do(req)
-		if err != nil {
-			t.Fatal("Do req", "failed to make request", err)
-			return -1, nil
-		}
+	requestAndExpect(t, p.URL, 10, http.StatusOK, nil)
 
-		defer rsp.Body.Close()
-		_, err = ioutil.ReadAll(rsp.Body)
-		if err != nil {
-			t.Fatal("read", "failed to read response", err)
-		}
+	expectHeader := strconv.Itoa(ratelimitSettings.MaxHits * int(time.Hour/ratelimitSettings.TimeWindow))
 
-		return rsp.StatusCode, rsp.Header
-	}
+	requestAndExpect(t, p.URL, 10, http.StatusTooManyRequests, http.Header{ratelimit.Header: []string{expectHeader}})
 
-	for i := 0; i < 10; i++ {
-		d0 := doc(128)
-		code, _ := request(d0)
-		if code == http.StatusTooManyRequests {
-			t.Fatal("should not be ratelimitted")
-		}
-	}
-	for i := 0; i < 10; i++ {
-		d0 := doc(128)
-		code, header := request(d0)
-		if code != http.StatusTooManyRequests {
-			t.Fatal("should be ratelimitted")
```

Comment:
E.g. [this failure](https://github.com/zalando/skipper/pull/1670/checks?check_run_id=1622366041) is apparenly due to slow `doc(128)` which uses `rand.Read`. Note also that doc is created but is never used